### PR TITLE
python modernize 2to3 tool fix_filter

### PR DESCRIPTION
**Pull request by davidlange6**
apply fixes.fix_filter tool from python futurize tool

### CODE REVIEW

File: FWCore/GuiBrowsers/python/Vispa/Gui/ConnectableWidget.py [hunk 1]
```diff
@@ -289,15 +289,15 @@ def sinkPorts(self):
         return [port for port in self._ports if port.portType() == "sink"]
         def isSink(port):
             return port.portType() == 'sink'
-        return filter(isSink, self._ports)
+        return list(filter(isSink, self._ports))
```
Comment (Dr15Jones) [inline FWCore/GuiBrowsers/python/Vispa/Gui/ConnectableWidget.py]: Do we understand this change? Does `filter` return a generator in python 3 while the old `filter` returned a list?

**Comment by davidlange6** [comment]
Right.

https://stackoverflow.com/questions/41666977/python-2-vs-python-3-difference-in-behavior-of-filter

**Comment by davidlange6** [comment]
please test

**Comment by civanch** [comment]
+1

**Comment by dmitrijus** [comment]
+1

**Comment by arunhep** [comment]
+1

**Comment by perrotta** [comment]
+1
- Technical
- Jenkins tests pass

**Comment by prebello** [comment]
+1

**Comment by thomreis** [comment]
+1

**Comment by fabiocos** [comment]
+operations

the expanded output of the Configuration/DataProcessing unit test is unchanged

**Comment by ggovi** [comment]
+1

File: FWCore/GuiBrowsers/python/Vispa/Gui/ConnectableWidget.py (hunk 1 above)
Comment (davidlange6) [reply FWCore/GuiBrowsers/python/Vispa/Gui/ConnectableWidget.py]: so now I understand you consider this a problem. not sure what alternatives there are. I'll have a look

Comment (davidlange6) [reply FWCore/GuiBrowsers/python/Vispa/Gui/ConnectableWidget.py]: seems an easy replacement is 

return [item for item in self._ports if isSink(item)]

I'd propose to handle the 13 changes (mostly in obsolete code and all in non-performance critical code) for a followup pr.

Comment (Dr15Jones) [reply FWCore/GuiBrowsers/python/Vispa/Gui/ConnectableWidget.py]: I concur.

**Comment by Dr15Jones** [comment]
+1

**Comment by fabiocos** [comment]
+1

**Comment by fabiocos** [comment]
merge

**Pull request merged**